mne/annotations.py

```diff
@@ -303,6 +303,44 @@ def read_annotations(fname):
     return annotations
 
 
+def read_brainstorm_annotations(fname, orig_time=None):
+    """Read annotations from a Brainstorm events_ file.
+
+    Parameters
+    ----------
+    fname : str
+        The filename
+    orig_time : meas_date
+        The 0 timestamp 
+
+    Returns
+    -------
+    annot : instance of Annotations | None
+        The annotations.
+    """
+    from scipy import io
+    list_annot = []
+    onsets = []
+    durations = []
+    descriptions = []
+    annot_data = io.loadmat(fname)
+
+    for label, _, _, _, times, _, _ in annot_data['events'][0]:
+        n_annot = len(times[0])
+        onsets.append(times[0])
+        if times.shape[0] == 2:
```

Comment:
to me, this should be a function.
```py
if is_event(times):
    durations.append(np.zeros(n_annot))
else
    durations.append(times[1] - times[0])
```

But don't write a function in python if you call it only once. (We can open a long discussion if its more desirable readability or the cost of calling a function in python which is not free)

Long story short, `times.shape[0] == 2` its really domain specific and by reading the code in two weeks I would not know why am I using this condition. So either you add a comment here, or describe a bit how times is structured, etc..

Reply:
another option is to encapsulate the `if` itself. Inside something like `get_duration_from_times`.
```py
def read_brainstorm_annotations(fname, orig_time=None):
    def get_duration_from_times(t):
        if t.shape[0] == 2:
            durations.append(t[1] - t[0])
        else:
            durations.append(np.zeros(n_annot))
    

    annot_data = io.loadmat(fname)
    list_annot, onsets, durations, descriptions = ([], [], [], [])
    for label, _, _, _, times, _, _ in annot_data['events'][0]:
        n_annot = len(times[0])
        onsets.append(times[0])
        durations.append(get_duration_from_times(times))
        descriptions += [str(label[0])] * n_annot

    return Annotations(onsets=np.concatenate(onsets),
                       durations=np.concatenate(durations),
                       descriptions=descriptions,
                       orig_time=orig_time)

```